rust/kernel/endian.rs

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Endian integer types.
+//!
+//! C header: [`include/uapi/linux/types.h`](../../../../include/uapi/linux/types.h)
+
+use crate::static_assert;
+
+macro_rules! define_le_integer {
+    ($name:ident, $native_type:ty) => {
+        #[doc = concat!("Represents little-endian integer `__", stringify!($name), "`.")]
+        /// # Examples
+        ///
+        /// ```
+        #[doc = concat!("use kernel::endian::", stringify!($name), ";")]
+        ///
+        #[doc = concat!("let n = 0x1A", stringify!($native_type), ";")]
+        #[doc = concat!("let v = ", stringify!($name), "::from(n);")]
+        #[doc = concat!("assert_eq!(", stringify!($native_type), "::from(v), n);")]
+        /// ```
+        #[allow(non_camel_case_types)]
+        #[derive(Default, Clone, Copy)]
+        #[repr(transparent)]
+        pub struct $name($native_type);
+
+        impl From<$native_type> for $name {
+            #[inline(always)]
+            fn from(v: $native_type) -> Self {
+                Self(<$native_type>::from_le(v))
```

Comment:
Maybe use to_le for $native_type -> Self? Both methods have the same definition, but semantically using to_le makes more sense here.

Reply:
Surely. Updated both le and be conversions.